main.go

```diff
@@ -58,24 +64,47 @@ func run() int {
 	return 0
 }
 
-func createStep(logger log.Logger) step.XcodeTestRunner {
+func createConfigParser(logger log.Logger) step.XcodeTestConfigParser {
 	envRepository := env.NewRepository()
+	commandFactory := command.NewFactory(envRepository)
 	inputParser := stepconf.NewInputParser(envRepository)
-	xcprettyInstaller := xcpretty.NewInstaller(goxcpretty.NewXcpretty(logger), logger)
+	xcodeVersionReader := xcodeversion.NewXcodeVersionReader()
+	pathModifier := pathutil.NewPathModifier()
+	deviceFinder := destination.NewDeviceFinder(logger, commandFactory)
+	utils := step.NewUtils(logger)
+
+	return step.NewXcodeTestConfigParser(inputParser, logger, xcodeVersionReader, deviceFinder, pathModifier, utils)
+}
+
+func createStep(logger log.Logger, logFormatter string) (step.XcodeTestRunner, error) {
+	envRepository := env.NewRepository()
 	commandFactory := command.NewFactory(envRepository)
 	pathChecker := pathutil.NewPathChecker()
 	pathProvider := pathutil.NewPathProvider()
 	pathModifier := pathutil.NewPathModifier()
 	fileManager := fileutil.NewFileManager()
 	xcconfigWriter := xcconfig.NewWriter(pathProvider, fileManager, pathChecker, pathModifier)
-	xcodebuilder := xcodebuild.NewXcodebuild(logger, commandFactory, pathChecker, fileManager, xcconfigWriter)
-	deviceFinder := destination.NewDeviceFinder(logger, commandFactory)
 	simulatorManager := simulator.NewManager(logger, commandFactory)
 	swiftCache := cache.NewSwiftPackageCache()
 	testAddonExporter := testaddon.NewExporter(testaddon.NewTestAddon(logger))
 	stepenvRepository := stepenv.NewRepository(envRepository)
 	outputExporter := output.NewExporter(stepenvRepository, logger, testAddonExporter)
 	utils := step.NewUtils(logger)
 
-	return step.NewXcodeTestRunner(inputParser, logger, xcprettyInstaller, xcodebuilder, deviceFinder, simulatorManager, swiftCache, outputExporter, pathModifier, pathProvider, utils)
+	xcodeCommandRunner := xcodecommand.NewRawCommandRunner(logger, commandFactory)
```

Comment:
nit: I find it a bit confusing that this value gets reassigned if `logFormatter` is `xcpretty`, a switch statement would make it easier to read.

Reply:
Will fix in follow-up PR.